app/types/app_version_type.rb

```diff
@@ -6,7 +6,7 @@ def cast(value)
   end
 
   def serialize(value)
-    value.to_pg_array if value
+    AppVersion.new(value).to_pg_array if value
```

Comment:
Turns out that find_or_initialize_by doesn't flow through cast before hitting serialize. Luckily AppVersion initializes just fine from an AppVersion so no big deal.